tests/test_move_embargo_args.py

```diff
@@ -28,55 +42,105 @@ def is_it_there(
             "--embargohours",
             str(embargo_hours),
             "--datasettype",
-            "raw",
+            *iterable_datasettype,
             "--collections",
-            "LATISS/raw/all",
+            # "LATISS/raw/all",
+            *iterable_collections,
             "--nowtime",
             now_time_embargo,
             "--move",
             move,
             "--log",
             log,
-        ]
+            "--desturiprefix",
+            desturiprefix,
+        ],
+        check=True,
     )
     # first test stuff in the temp_to butler
     butler_to = Butler(temp_to)
     registry_to = butler_to.registry
-    ids_in_temp_to = [
-        dt.dataId.full["exposure"]
-        for dt in registry_to.queryDatasets(datasetType=..., collections=...)
-    ]
-    # verifying the contents of the temp_to butler
-    # check that what we expect to move (ids_should_be_moved)
-    # are in the temp_to repo (ids_in_temp_to)
-    assert sorted(ids_should_be_moved) == sorted(
-        ids_in_temp_to
-    ), f"{ids_should_be_moved} should be in {temp_to} repo but isnt :(, \
-        what is in it is: {ids_in_temp_to}"
-    # now check the temp_from butler and see what remains
-    butler_from = Butler(temp_from)
-    registry_from = butler_from.registry
-    ids_in_temp_from = [
-        dt.dataId.full["exposure"]
-        for dt in registry_from.queryDatasets(datasetType=..., collections=...)
-    ]
+    counter = 0
+    for dtype in datasettype:
+        try:
+            if any(
+                dim in ["exposure", "visit"]
+                for dim in registry_to.queryDatasetTypes(dtype)[0].dimensions.names
+            ):
+                print("dtype with exposure or visit info: ", dtype)
+                ids_in_temp_to = [
+                    dt.dataId.mapping["exposure"]
+                    for dt in registry_to.queryDatasets(
+                        datasetType=..., collections=...
+                    )
+                ]
+            else:
+                print("dtype with no exposure", dtype)
+                datasetRefs = registry_to.queryDatasets(
+                    datasetType=datasettype, collections=collections
+                )
+                ids_in_temp_to = [dt.id for dt in datasetRefs]
+        except IndexError:
+            ids_in_temp_to = []
+
+        # verifying the contents of the temp_to butler
+        # check that what we expect to move (ids_should_be_moved)
+        # are in the temp_to repo (ids_in_temp_to)
+        assert sorted(ids_should_be_moved) == sorted(
```

Comment:
The default pytest assert output should list the mismatches and is easier to maintain.

Reply:
Are you suggesting removing the f statement following? @ktlim Because when I do that I don't get to see the full list of expected IDs.

Reply:
I looked into this a little more. I'm going to keep the more verbose option for some of these because it is the only way I get the printout for the full list of IDs in each list:
```
E           AssertionError: [2019111300059, 2019111300061, 2020011700002, 2020011700003, 2020011700004] should be in /tmp/tmpfjok7hra/temp_test_to repo but isnt :(,             what is in it is: [2019111300059, 2019111300061, 2020011700002, 2020011700003]
E           assert [201911130005...2020011700004] == [201911130005...2020011700003]
E             Left contains one more item: 2020011700004
```
Without the F statement I get:
```
E           assert [201911130005...2020011700004] == [201911130005...2020011700003]
E             Left contains one more item: 2020011700004
```